### PR TITLE
[doc] Update documentation regarding minimum-priority

### LINKED ISSUE #4454: [core] "Unknown option: '-min'"  but is referenced in documentation
**Affects PMD Version:** 

7.0.0-rc1

**Description:**

The `-min` parameter is no longer allowed. I cannot use the priority (1-5) for rules to exclude some from being checked.

`--minimum priority` is a Low/Medium/High value, so that can not be the solution:
```
Rule priority threshold; rules with lower priority than configured here won't be used. Valid values (case insensitive): High, Medium High, Medium, Medium Low, Low
```

Substituting --minimum-priority gives a long error message basically saying that it is not a number, but a string.

[-min is referenced in documentation](https://docs.pmd-code.org/latest/pmd_userdocs_configuring_rules.html#message-and-priority-overriding)

```
Run ~/pmd/bin/pmd check -d force-app -d sample-handler --rulesets pmd/ruleset.xml --format csv --no-cache -min 2 --verbose -r pmd/pmd.csv
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/runner/pmd/lib/groovy-2.[4](https://github.com/dschach/salesforce-trigger-framework/actions/runs/4622271015/jobs/8174740004#step:4:5).21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Unknown option: '-min'
Possible solutions: --minimum-priority
Error: Process completed with exit code 2.
```

**Code Sample demonstrating the issue:**

```bash
~/pmd/bin/pmd check --dir force-app --dir sample-handler --rulesets pmd/ruleset.xml --format csv --no-cache -min 2 --verbose -r pmd/pmd.csv
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

Run pmd with `-min` parameter and 

**Running PMD through:** *GitHub Action*: https://github.com/dschach/salesforce-trigger-framework/blob/main/.github/workflows/ci.yml

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

- [x] Add a unit test for `--minimum-priority`. Figure out, how to deal with the whitespace in "Medium Low". 
- [x] Maybe support additionally a numeric value for the priority? Since we use numbers in the ruleset, it makes sense...

## Related issues

- Fixes #4454

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 10 violations,<br />
introduces 1 new violations, 0 new errors and 0 new configuration errors,<br />
removes 1 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/c8bcae07d8c54c65840a6ebd49d34bcf/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by adangel** [comment]
This is ready now. I've decided to do the following:
- implement an own RulePriorityTypeSupport class to convert the CLI parameter. This allows us to have more control on what we support: We support now numeric values as well as the enum values (both name() and getName()).
  - I think this make sense, since we supported it in PMD 6 and in the ruleset, only numeric values are allowed. So it makes sense to still allow numeric values.
  - Future: #4327
- use name() for the completion candidates, as they don't have a space in it - then it's less hassle to use e.g. "medium_high". Otherwise it depends on your shell, whether/how you need to escape the space. Using the name with the space ("Medium High") is still supported.

File: docs/pages/pmd/userdocs/extending/rule_guidelines.md [hunk 1]
```diff
@@ -12,26 +12,37 @@ Here is a bunch of thing to do you may consider once your rule is “up and runn
...
-2.  **Change highly recommended.** Behavior is quite likely to be broken/buggy.
-3.  **Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
-4.  **Change optional.** Behavior is not likely to be buggy, but more just flies in the face of standards/style/good taste.
-5.  **Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
+1.  **High: Change absolutely required.** Behavior is critically broken/buggy.
+2.  **Medium High: Change highly recommended.** Behavior is quite likely to be broken/buggy.
+3.  **Medium: Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
+4.  **Medium Low: Change optional.** Behavior is not likely to be buggy, but more just flies in the face of
+    standards/style/good taste.
+5.  **Low: Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
```
Comment (jsotuyod) [inline docs/pages/pmd/userdocs/extending/rule_guidelines.md:23]: notice almost all our rules are priority 3 (Medium) by default, we don't really follow this ourselves

Comment (dschach) [reply docs/pages/pmd/userdocs/extending/rule_guidelines.md:23]: While that's true, in my projects, I like to assign my own priority value for each rule so that my developers know what to focus on first... and then I can put a `minimum-priority` flag in the CLI and ignore lower-priority items.

Comment (adangel) [reply docs/pages/pmd/userdocs/extending/rule_guidelines.md:23]: related issues: #4327, #3926
